src/js/media-captions-button.js

```diff
@@ -136,6 +140,41 @@ class MediaCaptionsButton extends MediaChromeButton {
     ) {
       updateAriaChecked(this);
     }
+    if (
+      this.hasAttribute('default-showing') && // we want to show captions by default
+      this.getAttribute('aria-checked') !== 'true' // and we aren't currently showing them
+    ) {
+      // Make sure we're only checking against the relevant attributes based on whether or not we are using subtitles fallback
+      const subtitlesIncluded = !this.hasAttribute('no-subtitles-fallback');
+      const relevantAttributes = subtitlesIncluded
+        ? [
+            MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+            MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+          ]
+        : [MediaUIAttributes.MEDIA_CAPTIONS_LIST];
+      // If one of the relevant attributes changed...
+      if (relevantAttributes.includes(attrName)) {
+        // check if we went
+        // a) from captions (/subs) not ready to captions (/subs) ready
+        // b) from captions (/subs) ready to captions (/subs) not ready.
+        // by using a simple truthy (empty or non-empty) string check on the relevant values
+        // NOTE: Need to get the attributes from the instance, as there may be more than one that's relevant
+        const nextCaptionsReady = subtitlesIncluded
```

Comment:
Optional: This might be easier to read without the template string (though a clever approach).
```
const nextCaptionsReady = !!this.getAttribute(MediaUIAttributes.MEDIA_CAPTIONS_LIST)
  || !!(subtitlesIncluded && this.getAttribute(MediaUIAttributes.MEDIA_SUBTITLES_LIST));
```